source/extensions/filters/common/ext_authz/ext_authz_http_impl.cc

```diff
@@ -57,72 +59,108 @@ struct SuccessResponse {
   ResponsePtr response_;
 };
 
-std::vector<Matchers::LowerCaseStringMatcherPtr>
-createLowerCaseMatchers(const envoy::type::matcher::v3::ListStringMatcher& list) {
-  std::vector<Matchers::LowerCaseStringMatcherPtr> matchers;
+envoy::type::matcher::v3::StringMatcher
+ignoreCaseStringMatcher(const envoy::type::matcher::v3::StringMatcher& matcher) {
+  envoy::type::matcher::v3::StringMatcher ignore_case;
+  ignore_case.set_ignore_case(true);
+  switch (matcher.match_pattern_case()) {
+  case envoy::type::matcher::v3::StringMatcher::MatchPatternCase::kSafeRegex:
+    return matcher;
```

Comment:
the `ignore_case` only supports prefix/suffix/exact matching and it's disallowed to use with regex/safe_regex: https://github.com/envoyproxy/envoy/blob/c472e94fee6e30022f35f46c8e98ca3b5cd8dfe9/source/common/common/matchers.cc#L75-L77

Maybe we should extend the `ignore_case` to also support regex/safe_regex if that is a valid use case?

Reply:
Thanks @yangminzhu for the heads up. I should’ve be more careful. I think we’re fine for now. But yeah extending that does make sense.